walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,107 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    /**
+     * Given the duration for a meeting, a set of attendees are required
+     * to attend the meeting, and a set of events with their own respective 
+     * set of attendees, find and return the set of valid meeting times. 
+     * Valid meeting times are time ranges such that the meeting time is 
+     * greater than the given duration and mandatory attendees are available 
```

Comment:
The function returns the valid meeting times for all *attendees*, right?
The 'mandatory', 'required' words are bit confusing to me, since this function only ones the set of given attendees and has no idea of 'mandatory' nor 'optional' attendees. It would be clear to write the comment based on input argument here.